crates/runtime/src/dataconnector/https.rs

```diff
@@ -81,12 +189,26 @@ impl DataConnectorFactory for HttpsFactory {
         &self,
         params: ConnectorParams,
     ) -> Pin<Box<dyn Future<Output = super::NewDataConnectorResult> + Send>> {
-        Box::pin(async move {
-            Ok(Arc::new(Https {
-                params: params.parameters,
-                tokio_io_runtime: params.io_runtime,
-            }) as Arc<dyn DataConnector>)
-        })
+        let file_format = params
+            .parameters
+            .get("file_format")
+            .expose()
+            .ok()
+            .map(|s| s.to_ascii_lowercase());
+
+        match file_format.as_deref() {
+            Some("json") | Some("jsonl") | Some("auto") => Box::pin(async move {
+                Ok(Arc::new(Https {
+                    params: params.parameters,
+                }) as Arc<dyn DataConnector>)
+            }),
+            _ => Box::pin(async move {
+                Ok(Arc::new(HttpListingConnector::new(
+                    params.parameters,
+                    params.io_runtime,
+                )) as Arc<dyn DataConnector>)
+            }),
+        }
```

Comment:
This is what determines if we treat HTTP datasets as before (HTTP as object store) and the new functionality (as path, query, body)
```rust
match file_format.as_deref() {
  // ...
```